### ISSUE TITLE
Geolocation - Background location tracking

### ISSUE THREAD

**Issue by BillyMartin1964** [xamarin/Essentials#297]
The current geolocation api has the ability to grab the current location, which is great, but would be nice to also get location changes and tracking. This has to be well thought out as if they are when the device is open, closed, and proper permissions and features. The Geolocation Plugin has this ability however it has morphed over time and has a lot of features (James is currently editing this post and is the creator so I am judging my own package lol). 

//Original:
Can you add in all of the features that James had? I tried to switch over and the new plugin didn't seem to do half as much. I though it would be an easy transfer for my Tracking App, but ended up being a huge task and after hours, I switched back. ):

**Comment by Redth** [comment]
Which particular features do you feel are missing?

**Comment by jamesmontemagno** [comment]
I have updated it with more information

**Comment by BillyMartin1964** [comment]
Thank you so much,  @jamesmontemagno . Since everything switched to .net I feel like I am swimming against a rip tide. ALL of your plugins were so great, and worked so well.  I used many of them, but now that I'm trying to modernize an already published app, it just makes me wonder if .Net is worth it.

I'm sure it's just growing pains. I decided to just start a new XF .net solution and bring in all of my pages and files, but before I had even started to import, I tried to install all of the nuget packages. Xamarin Essentials would not install above 0.6.0 due to Android support files only being at 25.4.0.2, but it wouldn't let me update the support files nor the Essentials Package... and this was a brand new solution with nothing in it. And before you ask, I had already changed the minimum and target builds. (:

**Comment by jamesmontemagno** [comment]
Simply install the two dependencies for Android first which is Core.Utils and CustomTabs. This is just Android support packages. This would be with any NuGet package that depends on any nuget package. The error message informs you to install the dependencies manually and match the version numbers. We have added guidance in the getting started docs. 

Dependencies are something we are actively working on though as you manually edit gradle files over in the world of Android and easily run into issues over there too.

**Comment by BillyMartin1964** [comment]
Thanks for the advice @jamesmontemagno ! It would only let me install the dependencies at version 25.4.0.2, but that was enough to get the essentials updated... THEN it let me update the android support packages all the way to 27.0.2.1. Weird, hunh? Anyway, thanks for your continued help. I can see why Microsoft hired you!

**Comment by jamesmontemagno** [comment]
@BillyMartin1964 we are adding some additional guidance on this in docs as it is a complicated thing with package references, but here is what one of my updates looked like to an app:

https://github.com/jamesmontemagno/app-compass/commit/fa15a9e2e80ed86cb3a1233118c34942a5f44834

**Comment by jamesmontemagno** [comment]
Duplicate of #290

**Comment by TonyLugg** [comment]
Not sure why James would have difficulty adding background tracking as he built it into his Geolocator plugin.  Looking at the Essentials Geolocation code it doesn't look like it would be a stretch to add a watch feature.

**Comment by jamesmontemagno** [comment]
I would say that my implementation on my plugin is very poor and doesn't handle proper backgrounds on any platform correctly. We would be going down the route of adding location updates from essentials, but from the foreground.  Background takes so much customization per platform that it would be impossible to get it right as I surely didn't.

**Comment by TonyLugg** [comment]
Just a thought.  If we only wanted to grab the location every few minutes (say every 10 minutes), could we not simply use Essentials Geolocation.GetLocationAsync from a background task?

**Comment by jamesmontemagno** [comment]
Potentially yes, nothing stopping you from doing that. Just check to make sure you have permissions before hand.

**Comment by TonyLugg** [comment]
Doesn't the Geolocation component automatically check the permissions?

**Comment by jamesmontemagno** [comment]
it does, but it would prompt on the UI thread...

**Comment by TonyLugg** [comment]
We would check the permissions before starting the background task.  If no permission, we would not start the task.  Within the task we can assume the user already granted required permissions (if we were able to grab the current position in the UI).  Or is there more that is required?

**Comment by TonyLugg** [comment]
@BillyMartin1964 This is the Xamarin.Essentials repo, not Geolocator plugin.  You should post there.

**Comment by BillyMartin1964** [comment]
I know what it is, @TonyLugg . That's why I started this thread. To ask for all of the features James (@jamesmontemagno ) had in his locator plugin to be put into Essentials. Please read the entire thread if you have time.
